packages/aft/lib/src/test_reports/test_report_folio.dart

```diff
@@ -0,0 +1,222 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:io';
+
+import 'package:aft/src/models.dart';
+import 'package:aft/src/test_reports/test_report_pass_fail.dart';
+import 'package:aft/src/test_reports/test_report_scored.dart';
+import 'package:aft/src/test_reports/test_score.dart';
+import 'package:aft/src/utils/constants.dart';
+import 'package:aft/src/utils/emphasize_text.dart';
+
+import 'test_report.dart';
+
+enum TestType {
+  integVM,
+  integWeb,
+  unit,
+}
+
+/// Encapsulates test information for display in the terminal
+class TestReportFolio {
+  TestReportFolio();
+
+  /// Specifies whether the test is a unit test or integration test
+  late TestType testType;
+
+  /// The test reports included in the folio
+  final List<TestReport> testReports = [];
+
+  /// Packages within the folio with test runs that exited with non-zero code
+  final List<PackageInfo> packegesWithExitExceptions = [];
+
+  List<TestReport> reportsByPackage(PackageInfo package) {
+    final reports =
+        testReports.where((report) => report.package == package).toList();
+
+    if (reports.isEmpty) {
+      stderr.writeln(
+        formatException(
+          'reportsByPackage: No reports found for ${package.name}.',
+        ),
+      );
+    }
+    return reports;
+  }
+
+  List<PackageInfo> availablePackages() {
+    final availablePackages = <PackageInfo>[];
+    for (final report in testReports) {
+      if (!availablePackages.contains(report.package)) {
+        availablePackages.add(report.package);
+      }
+    }
+    if (availablePackages.isEmpty) {
+      stderr.writeln(
+        formatException(
+          'availablePackages: No packages found in TestReportFolio.',
+        ),
+      );
+    }
+    return availablePackages;
+  }
+
+  List<String> availableFilesByPackage(PackageInfo package) {
+    final reports = reportsByPackage(package);
+    final fileNames = <String>[];
+    for (final report in reports) {
+      if (!fileNames.contains(report.fileName)) {
+        fileNames.add(report.fileName);
+      }
+    }
+    if (fileNames.isEmpty) {
+      stderr.writeln(
+        formatException(
+          'availableFilesByPackage: No files found for ${package.name}.',
+        ),
+      );
+    }
+    return fileNames;
+  }
+
+  TestReport reportByFile(PackageInfo package, String fileName) {
+    return testReports.firstWhere(
+      (report) => report.package == package && report.fileName == fileName,
+    );
+  }
+
+  /// Returns a test report by test id.
+  ///
+  /// Will only return a report for Flutter or Dart unit tests.
+  TestReport? reportByTestId(PackageInfo package, int testId) {
+    TestReport? report;
+    try {
+      report = testReports.firstWhere(
```

Comment:
Use `firstWhereOrNull` from `package:collection`

Reply:
Awesome, thanks for that.